nautobot/extras/filters.py

```diff
@@ -98,6 +100,110 @@ class CreatedUpdatedFilterSet(django_filters.FilterSet):
     last_updated__lte = django_filters.DateTimeFilter(field_name="last_updated", lookup_expr="lte")
 
 
+class RelationshipFilter(django_filters.Filter):
+    """
+    Filter objects by the presence of associations on a given Relationship.
+    """
+
+    def __init__(self, side, relationship=None, value_list=[], queryset=None, *args, **kwargs):
+        self.relationship = relationship
+        self.value_list = value_list
+        self.qs = queryset
+        self.side = side
+        super().__init__(*args, **kwargs)
+
+    def filter(self, qs, value):
+        # We are not using value here because it only contains the latest/last uuid in value_list and we have no use for a single value.
+        # We need the complete list of uuids to filter
+
+        # When getting value_list from empty DynamicModelChoiceField self.value_list = ['']
+        # When getting value_list from empty DynamicMultipleModelChoiceField self.value_list = []
+        # The conditional here accounts for both situations
+        if not self.value_list or "" in self.value_list:
+            # value_list is here to check whether a dropdown value has been entered
+            return super().filter(qs, value)
+        else:
+            if self.side == "source":
+                values = RelationshipAssociation.objects.filter(
+                    destination_id__in=self.value_list,
+                    source_type=self.relationship.source_type,
+                    relationship=self.relationship,
+                ).values_list("source_id", flat=True)
+            else:
+                values = RelationshipAssociation.objects.filter(
+                    source_id__in=self.value_list,
+                    destination_type=self.relationship.destination_type,
+                    relationship=self.relationship,
+                ).values_list("destination_id", flat=True)
+
+            if len(qs) == len(self.qs):
+                # If qs has not been modified from the original queryset, we take the intersection of the original queryset and the filtered queryset.
+                # Essentially the filtered queryset.
+                qs &= self.get_method(self.qs)(Q(**{"id__in": values}))
+            else:
+                # If qs has been modified from the original queryset, we take the union of the modified qs and the filtered queryset.
+                # Essentially modified qs and the filtered queryset combined.
+                qs |= self.get_method(self.qs)(Q(**{"id__in": values}))
```

Comment:
This feels off to me. Can you give examples (preferably in unit-test form :-) ) of these two if/else cases and the expected outcome of each?